### ISSUE TITLE
"creation time is not available on this platform currently"

### ISSUE THREAD

**Issue by senden9** [mgattozzi/rdb#1]
Hi!

I build my binary crate via `cargo +nightly build`. Then I tried to use rdb via `rdb sprit_bot`. Result:
```
$ rdb sprit_bot       
creation time is not available on this platform currently
```

Did I use rdb wrong? Or is this some kind of bug? Some infos to my system:

 * Linux 4.19.12-arch1-1-ARCH
 * Archlinux
 * Cargo version of my crate: cargo 1.33.0-nightly (0d1f1bbea 2018-12-19)
 * Cargo version I use to compile rdb: cargo 1.31.0 (339d9f9c8 2018-11-16)

**Comment by mgattozzi** [comment]
Nope! This is not your fault at all! This is indeed a bug. [This is the method](https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.created) used [here](https://github.com/mgattozzi/rdb/blob/53e43f614d9df208391e499dbdecfa9e7fe8be48/src/main.rs#L57). When I wrote this I wanted to guarantee we used the most recently created file. However, not all platforms have this (not the case on my Mac Mojave machine) so the correct fix is:

1. Try created at
2. If it fails first try [modified](https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.modified)
3. If that fails try [accessed]https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.accessed)
4. If that fails then we're out of luck here and print out the error

created is more of a guarantee we'll pick the right one, but unless people go through and touch this folder this fix should suffice and if they have none of those then 🤷‍♂️ We need some way to distinguish between hash names for files
